File: src/main/java/com/sequenceiq/submiter/JobSubmitter.java
package com.sequenceiq.submiter;

import com.sequenceiq.submiter.config.AmbariConfigService;
import org.apache.hadoop.conf.Configuration;

public class JobSubmitter {

    public static void main(String[] args){
        Configuration configuration = new AmbariConfigService(
                System.getProperty("ambariIp"), System.getProperty("ambariPort"),
                System.getProperty("ambariUsername"), System.getProperty("ambariPassword")).getConfiguration();

        // TODO: submissions
    }

    public void runScaldingJob(Configuration configuration, String[] args) {
        // TODO
    }

    public void runSparkJob(Configuration configuration, String[] args) {
        // TODO
    }
}

File: src/main/scala/com/sequenceiq/submitter/JobSubmitter.scala
package com.sequenceiq.submitter

import com.sequenceiq.submitter.spark.SparkRunner
import com.twitter.scalding.Tool
import org.apache.hadoop.conf.Configuration
import org.apache.hadoop.util.ToolRunner
import org.apache.hadoop.yarn.api.records.ApplicationReport
import org.apache.spark.SparkConf

object JobSubmitter {
  def main(args: Array[String]) {
    // TODO
  }

  def runScaldingJob(conf: Configuration, args: Array[String]): Unit = {
    ToolRunner.run(conf, new Tool, args)
  }

  def runSparkJob(conf: Configuration, sparkConf: SparkConf, clientArgs: Array[String], async: Boolean): ApplicationReport = {
    new SparkRunner().run(conf, sparkConf, clientArgs, async)
  }

}
